Engine/source/persistence/taml/taml.cpp

```diff
@@ -644,18 +644,20 @@ ImplementEnumType(_TamlFormatMode,
       // Fetch field count.
       const U32 fieldCount = fieldList.size();
 
-      ConsoleObject* defaultConObject;
-      SimObject* defaultObject;
+      ConsoleObject* defaultConObject = NULL;
+      SimObject* defaultObject = NULL;
       if (!getWriteDefaults())
       {
          // Create a default object of the same type
          defaultConObject = ConsoleObject::create(pSimObject->getClassName());
+         if (!defaultConObject)
+            return;
          defaultObject = dynamic_cast<SimObject*>(defaultConObject);
       
-         // ***Really*** shouldn't happen
-         if (!defaultObject)
-            return;
       }
+      // ***Really*** shouldn't happen
+      if (!defaultConObject || !defaultObject)
```

Comment:
defaultObject will be NULL if the class is derived from NetEvent. defaultConObject will be NULL if an invalid class is specified.

Perhaps a safer approach here would be to make some sort of ConsoleObject::create_simobject_class function which only creates classes if they are derived from SimObject. Perhaps it could be expanded to only allow certain class types to be constructed from a taml file.

Reply:
in the interest of letting lukas complete https://github.com/TorqueGameEngines/Torque3D/pull/1004 in a timely manner, we'll toss this in with a note that in the context it's used for, the method is for dumping simobjects to and form taml files, so niether scenario occurs. though we will likely want to circle back for cleanliness's sake